src/pair.ts

```diff
@@ -39,6 +42,19 @@ export abstract class Pair {
 	}
 	protected abstract swapExtraData(inputToken: Address): string;
 	public abstract outputAmount(inputToken: Address, inputAmount: BigNumber): BigNumber;
+
+	public snapshot(): Snapshot {
```

Comment:
these should be "abstract" types right?

Reply:
okay, will make them `abstract`
I only started coding typescript recently.